operator/operator.php

```diff
@@ -70,6 +82,19 @@ public function set_user_categories($forum_id)
 		// TODO
 		// 1.if user is registered, update the db with serialized array of collapsed category data ($this->collapsed_categories)
 		// 2.set their cookie too by calling set_cookie_categories()
+
+		if ($this->user->data['is_registered'])
+		{
+			$sql = 'UPDATE ' . USERS_TABLE . '
+				SET collapsible_categories = ' . $this->db->sql_escape(serialize($this->collapsed_categories)) . '
+				WHERE user_id = ' . (int) $this->user->data['user_id'];
+			$this->db->sql_query($sql);
```

Comment:
Any possible error that might occur, no matter how rare. We only want to return true if the update actually occurs.